src/solver.jl

```diff
@@ -120,6 +120,24 @@ solved_objective_value(flux_balance_analysis(model, ...))
 solved_objective_value(opt_model)::Maybe{Float64} =
     is_solved(opt_model) ? objective_value(opt_model) : nothing
 
+function get_solution(semantics::Val{Semantics}, model::AbstractMetabolicModel, opt_model) where {Semantics}
+    sem = Accessors.Internal.get_semantics(semantics)
+    isnothing(sem) && throw(DomainError(semantics, "Unknown semantics"))
+    (_, _, _, sem_varmtx) = sem
+    is_solved(opt_model) ? sem_varmtx(model)' * value.(opt_model[:x]) : nothing
+end
+
+get_solution(semantics::Symbol, model::AbstractMetabolicModel, opt_model) = get_solution(Val(semantics), model, opt_model)
+
+function get_solution_dict(semantics::Val{Semantics}, model::AbstractMetabolicModel, opt_model) where {Semantics}
+    sem = Accessors.Internal.get_semantics(semantics)
+    isnothing(sem) && throw(DomainError(semantics, "Unknown semantics"))
+    (ids, _, _, sem_varmtx) = sem
+    is_solved(opt_model) ? Dict(ids .=> sem_varmtx(model)' * value.(opt_model[:x])) : nothing
+end
+
+get_solution_dict(semantics::Symbol, model::AbstractMetabolicModel, opt_model) = get_solution_dict(Val(semantics), model, opt_model)
+
```

Comment:
This is great, thanks.

As a tuning opportunity, the `get_` prefixes implies an imperative extraction operation, which should be possibly avoided, we want the whole thing to feel as stateless as possible. `solution` is probably too short, maybe we go for something like `solution_variable_values` and `solution_reaction_values` etc? Other ideas welcome tho.

Reply:
will think of something (currently favoring `value`)